ReactiveFeedback/SignalProducer+System.swift

```diff
@@ -29,8 +29,23 @@ extension SignalProducer where Error == NoError {
 
             return SignalProducer<Event, NoError>(Signal.merge(events))
                 .scan(initial, reduce)
-                .prefix(value: initial)
                 .on(value: stateObserver.send(value:))
+                .prefix(value: initial)
+                .on(started: {
```

Comment:
TBH this doesn't look nice since `initial` is now used 3 times in this pipeline (ideally, it should only be used for "initial" setup once).

IMO the core problem is the initial feedback loop running **before system's initialization completes (regardless of which thread it had happened)**.
As this PR has just demonstrated, the reason is obviously initial (and too early) feedback passed in as an system's initializer argument, so we need to pay more attention to avoid this situation...

For now, I think it's good to have this PR accepted as a workaround.
I personally recommend to group `on(started:value:)` together so that we can recognize `on(started:)` is branched from `on(value:)`.

```swift
.on(
    started: { /* initial early feedback workaround */ }, 
    value: stateObserver.send(value:)
)
.prefix(value: initial)
```

Reply:
Also, we probably need to `state <~ SignalProducer.system(...).skip(first: 2)` instead since `initial` will be emitted twice.

Reply:
`prefix(value: initial)` sends value to observers
`stateObserver.send(value:)` sends value to `Feedbacks`

Reply:
@sergdort 
New pipeline shown in https://github.com/Babylonpartners/ReactiveFeedback/pull/38#discussion_r266434060 has a swapped order so `prefix(value: initial)` will no longer affect `stateObserver.send(value:)`.

Reply:
It will be sent there via `on(started: )`

```swift
SignalProducer<Event, NoError>(Signal.merge(events))
                .scan(initial, reduce)
                .on(value: stateObserver.send(value:))
                .prefix(value: initial)
                .on(started: {
                    // NOTE: Due to the nature of `prefix` lazily starting the producer being prefixed, we cannot rely
                    //       on `on(value:)` to ignite the feedbacks with the initial state.
                    //
                    //       At the time `prefix(value:)` calls `on(value:)` for the initial value, the events-reducer
                    //       producer has not yet been started yet. Consequentially, it would lead to dropped events
                    //       when the system is instantiated on a queue different from the queue used for
                    //       serializing events.
                    //
                    //       Having said that, `prefix(value:)` is guaranteed to have started the prefixed producer as
                    //       part of the synchronous producer starting process. So we can address the issue by applying
                    //       `on(started:)` after `prefix(value:)` to ignite the system, while having `on(value:)`
                    //       instead applied before `prefix(value:)` to keep the reducer-to-feedbacks path open.
                    stateObserver.send(value: initial)
                })
```